### DISCUSSION TITLE
How to get kitty tab title using kitty @ ls | jq

### DISCUSSION THREAD

**Discussion by salmankhilji** [kovidgoyal/kitty#3563] [Q&A]
The issue with the following `jq` query is that it _needs_ the tab to be _focused_ when it is called.  I have a bash script running in a _non-focused_ (i.e., background) tab that needs to get the title of the kitty tab its running in.

```
kitty @ ls | jq '.[0].tabs | .[] | select(.is_focused == true) | .title'
```

Writing a `jq `query that gets the tab title from within a background tab seems to be significantly harder.  Does kitty provider an easier way?

**Comment by kovidgoyal** [comment]
I dont use jq so I can help with that. If I were you I'd simply write a script in a proper programming language, which makes it trivial. Indeed just use python and you can run it with kitty itself, like this

kitty +launch myscript.py

**Comment by resolritter** [comment]
not sure if I understand the problem, but why don't you save the tab id in a variable when the tab is created? then you can use it in your jq script later

**Reply by salmankhilji** [reply]
> not sure if I understand the problem, but why don't you save the tab id in a variable when the tab is created? then you can use it in your jq script later

Indeed this is what I did to solve this problem. I saved the tab title into an environment variable that I could later access from within my script.  My initial idea to read it from `kitty @ ls` output would not have worked.  That is because even if a tab was created without the `--tab-title` option, the `kitty @ ls` output contained a fallback title, so the script did not know which title the user had specified.

Anyway, I refactored my scripts further to not even need this anymore.
